internal/command/workspace_select.go

```diff
@@ -95,9 +97,20 @@ func (c *WorkspaceSelectCommand) Run(args []string) int {
 		}
 	}
 
+	var newState bool
+
 	if !found {
-		c.Ui.Error(fmt.Sprintf(envDoesNotExist, name))
-		return 1
+		if orCreate {
+			_, err = b.StateMgr(name)
```

Comment:
This is a "Time of Check Time of Use" race condition, but the `workspace select` command doesn't have the ability to take a lock. I think this may need to incorporate the `new` command. I'll have to look into the other code path to see what that entails.

Reply:
@jbardin are there any changes needed?

Reply:
@jbardin Are changes actually needed here? or am I able to resolve this conversation?

Reply:
sorry, no changes needed here